Sources/PromiseKit/Any.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  Any.swift
+//  PovioKit
+//
+//  Created by Toni Kocjan on 26/08/2021.
+//  Copyright © 2020 Povio Labs. All rights reserved.
+//
+
+import Foundation
+
+/// Returns a new Promise, combining multiple promises.
+///
+/// Use this method when you need to combine the result of several promises of type `T`,
+/// where at least one has to succeed.
+///
+/// - Parameter promises: A collection of `Promises` that you want to combine.
+/// - Returns: An array of `Optional<T>` values wrapped in a Promise.
+public func any<T, C: Collection>(
+  on dispatchQueue: DispatchQueue? = .main,
+  promises: C
+) -> Promise<[T?]> where C.Element == Promise<T> {
+  guard !promises.isEmpty else {
+    return .error(NSError(domain: "com.poviokit.promisekit", code: 101, userInfo: nil))
+  }
+  
+  return .init { seal in
+    let barrier = DispatchQueue(label: "combineQueue", attributes: .concurrent)
+    for promise in promises {
+      promise.finally { result in
+        barrier.async(flags: .barrier) {
+          guard promises.allSatisfy({ $0.isResolved }) else { return }
```

Comment:
`isResolved` flags what?

Reply:
`isResolved` means is that the promise was fulfilled with a value.